### DISCUSSION TITLE
Change subproject delimeter

### DISCUSSION THREAD

**Discussion by trev-dev** [GothenburgBitFactory/taskwarrior#2779] [Q&A] [answered]
I do a lot of work on websites, and as such, like to enter projects by website domain.

If I run a `task projects` report, it tends to look a bit ugly because i use `:` as my project delimiter as opposed to a full stop `.`.

It ends up looking like this:
```
Project                 Tasks
(none)                     28
#website                    1
  com:Feature Addons        1
```

Instead of:
```
Project                 Tasks
(none)                     28
#website.com                1
  Feature Addons            1
```

The `#` is for metadata in a different report that I run with timewarrior, a report that I wrote to have the custom delimiter I am talking about.

I don't use the projects report that often, but this lack of flexibility concerns me if later features are missing flexibility, too.

**Comment by smemsh** [comment, marked as answer]
I would like to use `/` for hierarchy delimiters myself instead of `.`, but it is what it is.  In my experience patches are more than welcome though, if you feel like hacking on it to allow definition of delimiter from an rc-defined character.

**Reply by trev-dev** [reply]
Not a c programmer, but am very close to learning enough to contribute this. Famous last words are "how hard could that be?"

**Reply by tbabej** [reply]
I suspect it will be relatively non-trivial. The assumption that `.` is used as a (sub)project delimeter is used in various parts of the code base. Nothing a determined individual wouldn't be able to power through, but I suspect it's far from a "make a change to one place in the code base".

**Reply by trev-dev** [reply]
Unfortunately for me the path of least resistance was to open up my time/taskwarrior database files and :%s my way out of using a divergent delimiter. I appreciate your input @tbabej, and my task list is already full of burning red at the top ;)
